### DISCUSSION TITLE
Drop a node if it is at the beginning of a route

### DISCUSSION THREAD

**Discussion by jmintz4** [google/or-tools#3561] [Routing (and legacy CP) questions] [answered]
If a certain node is put at the beginning of a route, I would like to drop it and start the route at its next node.
How would I code this?
For example, I need to do something like:

idx_start = routing.Start(vehicle)   # the depot
j = 12
if routing.NextVar(idx_start) == j:
    drop j

**Comment by Regista6** [comment]
Create a counting dimension then constrain the cumul var of the node to not be 1?.

**Comment by jmintz4** [comment, marked as answer]
I have added constraints to prevent dummy nodes from being at the beginning or end of route, along with AddDisjunction to drop the nodes if they are "constrained away" (don't fit on any route).  But, no solution found.  Am I applying the disjunction with penalty correctly?
```py
    for node in range(1, (num_dummy_nodes + 1)):
        for veh in range(num_vehicles):

                idx_start = routing.Start(veh)
                start_node = sum([(routing.NextVar(idx_start) == j) \
                                  * j for j in range(1, num_nodes)])
                routing.solver().Add(start_node != node)

                idx_end = routing.End(veh)
                end_node = sum([(routing.NextVar(manager.NodeToIndex(j)) == manager.IndexToNode(idx_end)) \
                                * j for j in range(1, num_nodes)])
                routing.solver().Add(end_node != node)

    penalty = 0
    for node in range(1, (num_dummy_nodes + 1)):
        routing.AddDisjunction([manager.NodeToIndex(node)], penalty)
```

**Reply by jmintz4** [reply]
The above code is working for me now, dropping nodes as expected.  I corrected something elsewhere.

**Reply by Mizux** [reply]
```py
start_node = sum([(routing.NextVar(idx_start) == j) \
                                  * j for j in range(1, num_nodes)])
```
I would have written:
```py
start_node = sum([(routing.NextVar(idx_start) == manager.NodeToIndex(j)) \
                                  * j for j in range(1, num_nodes)])
```
